lib/internal/crypto/hash.js

```diff
@@ -26,16 +26,19 @@ const { normalizeEncoding } = require('internal/util');
 const { isArrayBufferView } = require('internal/util/types');
 const LazyTransform = require('internal/streams/lazy_transform');
 const kState = Symbol('kState');
-const kFinalized = Symbol('kFinalized');
 
 function Hash(algorithm, options) {
   if (!(this instanceof Hash))
     return new Hash(algorithm, options);
   validateString(algorithm, 'algorithm');
   legacyNativeHandle(this, new _Hash(algorithm));
-  this[kState] = {
-    [kFinalized]: false
-  };
+  Object.defineProperty(this, kState, {
+    value: {
+      finalized: false
+    },
+    configurable: true,
+    writable: true
+  });
```

Comment:
This is a pretty cheap entry and useful as debugging information, I’d keep it.

Reply:
It is always possible to view the entry when using `console.log(util.inspect(obj, { showHidden: true }))`. So I would rather keep it hidden from regular users and Node core developers should likely use that command to inspect things / define `util.inspect.defaultOptions.showHidden = true` in the REPL.

But I don't have a strong opinion about it and if you think it's best to be visible for everyone, I'll change it back.

Reply:
> So I would rather keep it hidden from regular users and Node core developers should likely use that command to inspect things / define `util.inspect.defaultOptions.showHidden = true` in the REPL.

What I meant, was: A lot of the information being hidden here is useful not only to Node.js core developers, even if we don’t encourage people to access it. That’s the point. :)

Reply:
May I ask in what way this information is valuable if it should not be accessed?

Reply:
> May I ask in what way this information is valuable if it should not be accessed?

Being able to inspect things for debugging, and to access them programmatically are pretty different things and I wouldn’t compare them?